### ISSUE TITLE
Introduce Scalar[T] concept with rank, shape, strides, offset defined

### ISSUE THREAD

**Issue by mratsim** [mratsim/Arraymancer#87]
Scalar types like float32 should be considered tensors of rank 0.

It would be very useful for backpropagation (loss function Scalar -> Tensor)
